File: python/LeetCode_1464_MaximumProductofTwoElementsinanArray.py
# Optimal solution

"""
def maxProduct(self, nums: List[int]) -> int:
    largest, second_Largest = 0, 0

    for num in nums:
        if num > largest:
        second_Largest = largest
        largest = num
        else:
            second_Largest = max(second_Largest, num)
    return (largest-1) * (second_Largest-1)
"""


def maxProduct(nums) -> int:
    n = sorted(nums)
    return (n[-1] - 1) * (n[-2] - 1)


print(maxProduct([3,4,5,2]))

File: c/LeetCode_1464_MaximumProductofTwoElementsinanArray.c
#include <stdio.h>
#include <stdlib.h>


int maxProduct(int* nums, int numsSize) {
    int max = 0;
    int max_2 = 0;

    for (int i = 0; i < numsSize; ++i) {
        if (nums[i] > max) {
            max_2 = max;
            max = nums[i];
        } else {
            max_2 = (max_2 > nums[i]) ? max_2 : nums[i];
        }
    }

    return (max - 1) * (max_2 - 1);
}


int main() {
    int nums[] = {10,2,5,2};
    int n = 4;

    printf("%d ", maxProduct(nums, n));

    return 0;
}